src/leveldown.h

```diff
@@ -66,6 +66,25 @@ static inline void DisposeStringOrBufferFromSlice(
   }                                                                            \
   leveldb::Slice to(to ## Ch_, to ## Sz_);
 
+#define LD_STRING_OR_BUFFER_TO_COPY(to, from, name)                            \
+  size_t to ## Sz_;                                                            \
+  char* to ## Ch_;                                                             \
+  if (!from->ToObject().IsEmpty()                                              \
+      && node::Buffer::HasInstance(from->ToObject())) {                        \
+    to ## Sz_ = node::Buffer::Length(from->ToObject());                        \
+    to ## Ch_ = new char[to ## Sz_];                                           \
+    memcpy(to ## Ch_, node::Buffer::Data(from->ToObject()), to ## Sz_);        \
+  } else {                                                                     \
+    v8::Local<v8::String> to ## Str = from->ToString();                        \
+    to ## Sz_ = to ## Str->Utf8Length();                                       \
+    to ## Ch_ = new char[to ## Sz_];                                           \
+    to ## Str->WriteUtf8(                                                      \
+        to ## Ch_                                                              \
+      , -1                                                                     \
+      , NULL, v8::String::NO_NULL_TERMINATION                                  \
+    );                                                                         \
+  }
+
```

Comment:
The cherry-pick behind this change had to be resolved, since we're applying this change on top of previous change to `LD_RUN_CALLBACK` macro below.